tests/python/gaia-ui-tests/gaiatest/apps/system/regions/cards_view.py

```diff
@@ -85,24 +88,22 @@ def close_app(self, app):
             expected.element_not_present(*self._app_card_locator(app)))
 
     def wait_for_cards_view(self):
-        cards_view = self.marionette.find_element(*self._task_manager_locator)
         Wait(self.marionette).until(
-            lambda m: cards_view.get_attribute('class') == 'active')
+            lambda m: self.root_element.get_attribute('class') == 'active')
 
     def wait_for_cards_view_not_displayed(self):
         Wait(self.marionette).until(
-            expected.element_not_displayed(*self._task_manager_locator))
+            expected.element_not_displayed(self.root_element))
 
     def swipe_to_previous_app(self):
         current_frame = self.apps.displayed_app.frame
 
-        start_x_position = current_frame.size['width']
-        final_x_position = start_x_position * 2
-        start_y_position = current_frame.size['height'] // 2
+        final_x_position = current_frame.rect['width'] // 2
+        start_y_position = current_frame.rect['height'] // 2
 
         # swipe forward to get previous app card
         Actions(self.marionette).flick(
-            current_frame, start_x_position, start_y_position, final_x_position, start_y_position).perform()
+            current_frame, 0, start_y_position, final_x_position, start_y_position).perform()
```

Comment:
I didn't really give much thought to it. This is something that works and is simpler compared to what we had (and which didn't work).